sdk/core/Azure.Core/tests/HttpPipelineRequestContentTests.cs

```diff
@@ -97,5 +97,33 @@ public void ObjectContent()
 
             CollectionAssert.AreEqual(expected, destination.ToArray());
         }
+
+        [Test]
+        public void DynamicDataContent()
+        {
+            ReadOnlySpan<byte> utf8Json = """
+                {
+                    "foo" : {
+                       "bar" : 1
+                    }
+                }
+                """u8;
+            ReadOnlyMemory<byte> json = new ReadOnlyMemory<byte>(utf8Json.ToArray());
+
+            JsonDocument doc = JsonDocument.Parse(json);
+            MemoryStream expected = new();
```

Comment:
Nit: tests, so less important, but you should generally using `using JsonDocument doc ...` et. al. for all disposables unless you know they shouldn't be disposed going out of scope (like your change in `RequestContent`).